data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/threshold/AbstractFlowThreshold.java

```diff
@@ -173,12 +182,20 @@ public boolean equals(Object o) {
             return false;
         }
         AbstractFlowThreshold threshold = (AbstractFlowThreshold) o;
-        return unit.equals(threshold.unit) && side.equals(threshold.side) && direction.equals(threshold.direction);
+        boolean result = true;
+        if (networkElement != null) {
+            result = networkElement.equals(threshold.networkElement);
+        }
+        return result && unit.equals(threshold.unit)
+            && side.equals(threshold.side) && direction.equals(threshold.direction);
```

Comment:
if networkElement is null and threshold.networkElement isn't null then this can still return true

Reply:
Should be fixed. I added a test.